web/models/CandidatoVagaDTO.php

```diff
@@ -17,9 +17,9 @@ public static function salvar($candidatoVaga)
     {
         $pdo = static::conectarDB();
 
-        if (empty($candidatoVaga->getUltimaDesistencia())) {
+        if (empty($candidatoVaga->getCandidato()->getId()) && empty($candidatoVaga->getVaga()->getId())) {
```

Comment:
Essa situação é um pouco peculiar, diferente do resto.
Por ser uma instancia relacional, não tem como ela nascer sem os IDs.

Isso só se aplicaria se estivéssemos falando do ID próprio, que é gerado na inserção do banco.
Mas nesse caso, essa tabela nao tem ID proprio. Ela usa IDs de outras tabelas, que sempre ja existem.

Então quando trabalhamos com relacional que não tem ID próprio, precisamos pensar outra maneira de identificar quando é insert ou update.
Nesse caso, a regra até então de verificar a ultima desistência fazia sentido, pq a mudança de status possível era somente da desistencia.

Agora, como temos outras mudanças, ela não vale.
Então o que precisamos checar é:

Se o status for o de Triagem e a ultimaDesistencia for vazia, é um insert.
Qualquer outra situação é um update.